src/Input.js

```diff
@@ -0,0 +1,61 @@
+import {
+  ERROR_MESSAGE_CAR_NAME_DUPLICATION,
+  ERROR_MESSAGE_CAR_NAME_INVALID,
+  ERROR_MESSAGE_USER_INPUT,
+  INPUT_MESSAGE_CAR_NAMES,
+  INPUT_MESSAGE_TRY_COUNT,
+} from './lib/constants.js';
+import {
+  checkArrayAllUnique,
+  readLineAsync,
+  splitIntoArray,
+  validatePositiveInteger,
+} from './lib/utils.js';
+
+class Input {
+  static #CAR_NAME_REGEXP = /^\w{1,5}$/;
+
+  static #SEPARATOR = ',';
+
+  #rawCars;
+
+  #rawTryCount;
+
+  constructor() {
+    this.#rawCars = '';
+    this.#rawTryCount = '';
+  }
+
+  async getUserInput() {
+    try {
+      this.#rawCars = await readLineAsync(INPUT_MESSAGE_CAR_NAMES);
+      this.#rawTryCount = await readLineAsync(INPUT_MESSAGE_TRY_COUNT);
+    } catch (e) {
+      throw new Error(ERROR_MESSAGE_USER_INPUT);
+    }
+  }
+
+  parseCars() {
+    const carArray = splitIntoArray(this.#rawCars, Input.#SEPARATOR);
+    Input.#validateCarArray(carArray);
+    return carArray;
+  }
+
+  parseTryCount() {
+    const tryCount = +this.#rawTryCount;
```

Comment:
number 형으로 캐스팅할 때 조금 더 명시적인 방법도 좋을 거 같아요!

```suggestion
    const tryCount = Number(this.#rawTryCount);
```

Reply:
[javascript/airbnb](https://github.com/airbnb/javascript?tab=readme-ov-file#coercion--numbers)에서도 위와 같은 방식(`Number`)을 권장하더라구요. 감사합니다 !